src/components/grader/submissions/SingleDiff.js

```diff
@@ -1,35 +1,134 @@
 import ReactDiffViewer from "react-diff-viewer";
-import React from "react";
-import AccordionSection from "@bit/nexxtway.react-rainbow.accordion-section";
-import { Label } from "semantic-ui-react";
+import React, { useState, useEffect } from "react";
+import { AccordionItem, Tile } from 'carbon-components-react';
+import CodeEditor from "./CodeEditor"
+import apiClient from "../../../api-client";
+
 
 function SingleDiff(props) {
+
+  const [theList, setTheList] = useState([])
+  const [visible, setVisible] = useState(false)
+
+
+  useEffect(() => {
+    apiClient.getFilesList(props.course, props.lab, props.result.id)
+      .then(res => {
+        setTheList(res.data)
+      });
+// eslint-disable-next-line
+  }, [])
+
+
+  useEffect(() => {
+   
+    if(theList.length > 0) {
+      setVisible(true)
+    }
+
+
+  }, [theList])
```

Comment:
what does this list do?

Reply:
* Holds list of file names for each submission at a time. It is required for the `CodeEditor` component
* Would it be better to rename it to `fileNameList`

Reply:
yes :smile: